src/main/webapp/app/exercise/feedback/feedback.component.ts

```diff
@@ -234,18 +240,17 @@ export class FeedbackComponent implements OnInit, OnChanges {
                     if (
                         this.result.assessmentType !== AssessmentType.AUTOMATIC_ATHENA &&
                         this.exerciseType === ExerciseType.PROGRAMMING &&
-                        this.result.submission?.participation &&
-                        (this.result.submission as ProgrammingSubmission).buildFailed
+                        (this.participation?.submissions?.[0] as ProgrammingSubmission)?.buildFailed
```

Comment:
Don't we need a length check before accessing the first element? Can we retrieve the element outside of the if check to keep the condition better readable?